open-apparel-registry/open-apparel-registry
# Create facilities tab & display text search results

## Overview

This PR adds a facilities list tab which will display a list of text search results as depicted below when the facilities results return from a search of the API. I also added a free text search which currently searches on the `name` `address` and `country_name` fields to narrow down the list of visible results, as well as a "Download CSV" functionality which downloads a list of CSV items.

Additionally, this PR also:

- replaces papaparse with a custom utility function to turn a 2d array into a string, which circumvents a bug whereby JSON.parse would sometimes try to parse the OAR ID as a number but fail and instead print `NaN` to the CSV
- adds tests for that newly added utility function
- adjusts the contributor list to list them in alphabetical order by name
- adjusts the app's DOM structure to make it possible to scroll subcomponents rather than scrolling the entire app
- changes the app name in `package.json` from "redpanda" (which was the legacy name?) to "open apparel registry"

Connects #209
Connects #227
Connects #127 
Connects #123 

## Demo

![facilities-list](https://user-images.githubusercontent.com/4165523/53747859-318c8f80-3e72-11e9-85a4-59ed271054f4.gif)

CSV: 
[facilities.csv.txt](https://github.com/open-apparel-registry/open-apparel-registry/files/2927348/facilities.csv.txt)

No results for a search:

![screen shot 2019-03-04 at 11 47 10 am](https://user-images.githubusercontent.com/4165523/53748396-474e8480-3e73-11e9-8807-6bbea0511f97.png)

## Notes

I made #244 for the map interactions because it will need some dedicated thought about how to handle the various possible kinds of interaction.

## Testing

- get this branch then `./scripts/update`
- run `./scripts/test` and verify that everything passes
- `./scripts/resetdb`
- visit the main map page and
    - verify that the contributors are now listed in alphabetical order by name
    - make a search and verify that you are navigated to the facilities tab on success
    - verify that the free text search works as expected
    - verify that you can download a list of facilities, filtered by your free text search
    - verify that you can scroll through the facilities and that clicking on any of the list items takes you to its details route
- on the Macbook-sized viewport, check that the following pages scroll correctly following the changes I made to the DOM:
   - `/auth/register`
   - `/contribute`
   - `/profile/:id/`
   - `/lists`
   - `/lists/:id`
- download a facility list and verify that that still works, following the changes to replace papaparse
